[MCPS] Add dashboards and widget tools to MCP Server toolsets docs (#35998)
DataDog/documentation: content/en/bits_ai/mcp_server/toolsets.md

- Find the Datadog user account for jane.doe@example.com.

### `get_datadog_dashboard`
*Toolset: **dashboards***\
*Required permissions: `DashboardsRead`, `UserAccessRead`*\
Retrieves a dashboard by ID, returning its title, description, tags, template variables, and full widget definitions.

- Get the full definition of dashboard `abc-def-ghi`.
- Show the widgets and template variables on the SRE overview dashboard.
- Retrieve dashboard details including all widget configurations for ID `xyz-123`.

### `upsert_datadog_dashboard`
*Toolset: **dashboards***\
*Required permissions: `DashboardsRead`, `DashboardsWrite`*\

> Creates a new dashboard or updates an existing one. Supports all widget types, template variables, and layout configuration. Auto-tags dashboards with `ai:created_with_ai` or `ai:modified_with_ai`.

Reviewer: "Auto-tags" is an informal compound verb; also, "new" is redundant (the word list flags "Create a new..." → "Create a...").

```suggestion
Creates a dashboard or updates an existing one. Supports all widget types, template variables, and layout configuration. Automatically tags dashboards with `ai:created_with_ai` or `ai:modified_with_ai`.
```

- Create a dashboard with timeseries widgets showing CPU and memory for `service:api`.
- Add a log stream widget to an existing dashboard `abc-def-ghi`.
- Build a dashboard with a top list of the top 10 services by error rate.

### `delete_datadog_dashboard`
*Toolset: **dashboards***\
*Required permissions: `DashboardsRead`, `DashboardsWrite`*\
Permanently deletes a dashboard by ID. This action is destructive and irreversible.

- Delete dashboard `abc-def-ghi`.
- Remove the test dashboard that was created earlier.

### `validate_dashboard_widget`
*Toolset: **dashboards***\
*Required permissions: `DashboardsRead` or `DashboardsWrite` or `NotebooksRead`*\

> Validates a widget definition against the dashboard schema. Call this when generating widget JSON to check for errors before creating or updating a dashboard.

Reviewer: "Call this" is an ambiguous pronoun reference. Adding "tool" resolves it.

```suggestion
Validates a widget definition against the dashboard schema. Call this tool when generating widget JSON to check for errors before creating or updating a dashboard.
```

- Validate this timeseries widget definition before adding it to a dashboard.
- Check if this query table widget JSON is correct.
- Verify the widget configuration built for a heatmap.

### `get_widget_reference`
*Toolset: **dashboards***\
*Required permissions: `DashboardsRead` or `DashboardsWrite` or `NotebooksRead`*\

> Retrieves schemas and building instructions for widget types. Returns TypeScript type definitions and step-by-step guidance. Call this before generating widgets to get the correct structure.

Reviewer: Same ambiguous pronoun as above.

```suggestion
Retrieves schemas and building instructions for widget types. Returns TypeScript type definitions and step-by-step guidance. Call this tool before generating widgets to get the correct structure.
```

> - How to build a timeseries widget?

Reviewer: "How to build..." is a sentence fragment. All other question-form examples in the file use complete grammatical questions.

```suggestion
- How do I build a timeseries widget?
```

- Show the schema for a top list widget.
- What fields are required for a query value widget?

### `ask_widget_expert`
*Toolset: **dashboards***\
*Required permissions: `DashboardsRead` or `DashboardsWrite` or `NotebooksRead`*\
Asks a Datadog widget expert a natural-language question. More token-efficient than `get_widget_reference` for targeted questions about widget configuration.

- How to add conditional formatting to a query table widget?
- What's the difference between a timeseries and a change widget?

> - How to use template variables in widget queries?

Reviewer: Same fragment pattern — "How to..." should be complete questions for consistency with the rest of the file.

```suggestion
- How do I add conditional formatting to a query table widget?
- What's the difference between a timeseries and a change widget?
- How do I use template variables in widget queries?
```
